app/master/subjob.py

```diff
@@ -27,10 +27,15 @@ def __init__(self, build_id, subjob_id, project_type, job_config, atoms):
         self.project_type = project_type
         self.job_config = job_config
         self._atoms = atoms
+        self._set_atoms_subjob_id()
```

Comment:
My only concern is that if someone accidentally changes the order in the constructor of the self._atoms = atoms and the self._set_atoms_subjob_id(), then the functionality will fail.

If you just write a unit test to make sure the atoms get the subjob id's set, then I think you're all good.